srcpkgs/orocos-kdl/template

```diff
@@ -1,17 +1,29 @@
 # Template file for 'orocos-kdl'
 pkgname=orocos-kdl
-version=1.4.0
+version=1.5.1
 revision=1
 wrksrc=orocos_kinematics_dynamics-${version}
 build_wrksrc=orocos_kdl
 build_style=cmake
-makedepends="eigen3.2"
+makedepends="eigen3.2 libcppunit-devel"
 short_desc="Orocos Kinematics and Dynamics C++ library"
 maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
 license="LGPL-2.1-only"
 homepage="http://www.orocos.org/kdl"
 distfiles="https://github.com/orocos/orocos_kinematics_dynamics/archive/v${version}.tar.gz"
-checksum=05b93e759923684dc07433ccae1e476d158d89b3c2be5079c20062406da7b4dd
+checksum=5acb90acd82b10971717aca6c17874390762ecdaa3a8e4db04984ea1d4a2af9b
+
+do_check() {
+	cd build
+	if [[ "$XBPS_LIBC" == "musl" ]]; then
+		## the __APPLE__ definition causes some typedefs that musl needs for tests.
+		cmake .. -DENABLE_TESTS:BOOL=ON -D 'CMAKE_CXX_FLAGS=-D__APPLE__=1'
+	else
+		cmake .. -DENABLE_TESTS:BOOL=ON
+	fi
+	ninja tests/all
+	ninja check
```

Comment:
This shouldn't be redoing the configuration. Instead these flags should be appended to `configure_args` before building...

Patching their usage of `__APPLE__` is probably our best bet, too.